README.md

```diff
@@ -152,3 +152,289 @@ commit it in this repo.
+configuration.
+
+### Custom `cardano-node` Version
+
+The configuration entry point is the `./flake.nix` file, which defines inputs
+and outputs for the deployment. Currently, the file includes local pins for
+customizations, as shown below:
+
+```nix
+  inputs = {
+    nixpkgs.follows = "cardano-parts/nixpkgs";
+    nixpkgs-unstable.follows = "cardano-parts/nixpkgs-unstable";
+    flake-parts.follows = "cardano-parts/flake-parts";
+    cardano-parts.url = "github:input-output-hk/cardano-parts";
+
+    # Local pins for additional customization:
+    cardano-node-8-12-2.url = "github:IntersectMBO/cardano-node/8.12.2";
+    iohk-nix-8-12-2.url = "github:input-output-hk/iohk-nix/577f4d5072945a88dda6f5cfe205e6b4829a0423";
+  };
+```
+
+To add a custom node version, simply include a new input pointing to the
+desired branch or revision. If you're using an older `cardano-node` version,
+you might also need to pin an appropriate version of `iohk-nix`.
+
+### Step 1: Modify `flake.nix`
+
+Add your custom inputs in the `flake.nix` file. For example, to add a custom
+`cardano-node` version:
+
+```nix
+  cardano-node-<version>.url = "github:<your-organization>/cardano-node/<version>";
+```
+
+If needed, also add the corresponding `iohk-nix` pin for compatibility with
+older versions.
+
+### Step 2: Update `colmena.nix`
+
+Next, update the `./flake/colmena.nix` file, which contains the configuration
+for each individual machine. Create a new variable for your custom node
+version, for example:
+
+```nix
+node8-12-2 = {
+  imports = [
+    config.flake.cardano-parts.cluster.groups.default.meta.cardano-node-service
+    inputs.cardano-parts.nixosModules.profile-cardano-node-group
+    inputs.cardano-parts.nixosModules.profile-cardano-custom-metrics
+    {
+      cardano-parts.perNode = {
+        lib.cardanoLib = config.flake.cardano-parts.pkgs.special.cardanoLibCustom inputs.iohk-nix-8-12-2 "x86_64-linux";
+        pkgs = {
+          inherit
+            (inputs.cardano-node-8-12-2.packages.x86_64-linux)
+            cardano-cli
+            cardano-node
+            cardano-submit-api
+            ;
+        };
+      };
+    }
+  ];
+};
+```
+
+This is based on the default `node` configuration, with custom inputs for the
+specified node version.
+
+### Step 3: Create Custom Configurations (Optional)
+
+If you don’t need to include previous `cardano-node` versions, you can
+simplify the configuration by omitting unused packages. For example:
+
+```nix
+node-tx-submission = {
+  imports = [
+    # Base cardano-node service
+    config.flake.cardano-parts.cluster.groups.default.meta.cardano-node-service
+
+    # Config for cardano-node group deployments
+    inputs.cardano-parts.nixosModules.profile-cardano-node-group
+    inputs.cardano-parts.nixosModules.profile-cardano-custom-metrics
+    {
+      cardano-parts.perNode = {
+        pkgs = {
+          inherit
+            (inputs.cardano-node-tx-submission.packages.x86_64-linux)
+            ;
```

Comment:
I think you meant to include the `cardano-node` package here for the tx-submission customization?

Reply:
oh do I have to add `cardano-node` here? :) I thought that line sufficed.

Reply:
Yes, because L402 - 404 as above are not inheriting anything, and so the default cardano-node package will still be the default defined for cardano-parts which won't be the tx-submission package.